src/silx/gui/plot/CompareImages.py

```diff
@@ -79,151 +81,297 @@ class AlignmentMode(enum.Enum):
         action.setIconVisibleInMenu(True)
         action.setCheckable(True)
         self.__autoAlignAction = action
         menu.addAction(action)
         if sift is None:
             action.setEnabled(False)
             action.setToolTip("Sift module is not available")
-        self.__alignmentGroup.addAction(action)
+        self.__group.addAction(action)
+
+    def getActionFromMode(self, mode: AlignmentMode) -> Optional[qt.QAction]:
+        """Returns an action from it's mode"""
+        for action in self.__group.actions():
+            actionMode = action.property("enum")
+            if mode == actionMode:
+                return action
+        return None
+
+    def setVisibleModes(self, modes: List[AlignmentMode]):
+        """Make visible only a set of modes.
+
+        The order does not matter.
+        """
+        modes = set(modes)
+        for action in self.__group.actions():
+            mode = action.property("enum")
+            action.setVisible(mode in modes)
+
+    def __selectionChanged(self, selectedAction: qt.QAction):
+        """Called when user requesting changes of the alignment mode.
+        """
+        self.__updateMenu()
+        mode = self.getSelected()
+        self.sigSelected.emit(mode)
+
+    def __updateMenu(self):
+        """Update the state of the action containing alignment menu.
+        """
+        selectedAction = self.__group.checkedAction()
+        if selectedAction is not None:
+            self.setText(selectedAction.text())
+            self.setIcon(selectedAction.icon())
+            self.setToolTip(selectedAction.toolTip())
+        else:
+            self.setText("")
+            self.setIcon(qt.QIcon())
+            self.setToolTip("")
+
+    def getSelected(self) -> AlignmentMode:
+        action = self.__group.checkedAction()
+        if action is None:
+            return None
+        return action.property("enum")
+
+    def setSelected(self, mode: AlignmentMode):
+        action = self.getActionFromMode(mode)
+        old = self.__group.blockSignals(True)
+        if action is not None:
+            # Check this action
+            action.setChecked(True)
+        else:
+            action = self.__group.checkedAction()
+            if action is not None:
+                # Uncheck this action
+                action.setChecked(False)
+        self.__updateMenu()
+        self.__group.blockSignals(old)
+
+
+class CompareImagesToolBar(qt.QToolBar):
+    """ToolBar containing specific tools to custom the configuration of a
+    :class:`CompareImages` widget
+
+    Use :meth:`setCompareWidget` to connect this toolbar to a specific
+    :class:`CompareImages` widget.
+
+    :param Union[qt.QWidget,None] parent: Parent of this widget.
+    """
+    def __init__(self, parent=None):
+        qt.QToolBar.__init__(self, parent)
+
+        self.__compareWidget = None
+
+        self.__visualizationToolButton = VisualizationModeToolButton(self)
```

Comment:
I would need to get some control over the different buttons / actions (__visualizationToolButton, __alignmentToolButton and __displayKeypoints) in order for example to hide some when needed.
Either by moving them to protected or even better by having getters

Reply:
I think for you it is better to create your own toolbar. So you are free to do what you want.
But you can reuse the sub components.

Reply:
If you realy want to reuse everything, i will propose to move that code into `silx.gui.plot.tools`m because the module start to be huge.

We can talk about it into another PR.